platform/security/handler/security-handler-basic/src/main/java/org/codice/ddf/security/handler/basic/WssBasicAuthenticationHandler.java

```diff
@@ -38,42 +37,58 @@
      */
     private static final String AUTH_TYPE = "WSSBASIC";
 
+    private final Parser parser;
+
+    public WssBasicAuthenticationHandler(Parser parser) {
+        this.parser = parser;
+    }
+
     protected BaseAuthenticationToken getBaseAuthenticationToken(String realm, String username,
             String password) {
         UsernameTokenType usernameTokenType = new UsernameTokenType();
         AttributedString user = new AttributedString();
         user.setValue(username);
         usernameTokenType.setUsername(user);
+        String usernameToken = null;
 
         // Add a password
         PasswordString pass = new PasswordString();
         pass.setValue(password);
         pass.setType(WSConstants.PASSWORD_TEXT);
         JAXBElement<PasswordString> passwordType = new JAXBElement<>(QNameConstants.PASSWORD,
-                PasswordString.class, pass);
-        usernameTokenType.getAny().add(passwordType);
+                PasswordString.class,
+                pass);
+        usernameTokenType.getAny()
+                .add(passwordType);
+
         // Marshall the received JAXB object into a DOM Element
-        String usernameToken = null;
-        Writer writer = new StringWriter();
-        try {
-            Set<Class<?>> classes = new HashSet<>();
-            classes.add(ObjectFactory.class);
-            classes.add(
-                    org.apache.cxf.ws.security.sts.provider.model.wstrust14.ObjectFactory.class);
+        List<String> ctxPath = new ArrayList<>(2);
+        ctxPath.add(ObjectFactory.class.getPackage()
+                .getName());
+        ctxPath.add(org.apache.cxf.ws.security.sts.provider.model.wstrust14.ObjectFactory.class.getPackage()
+                .getName());
+        if (null == parser) {
+            throw new IllegalStateException("XMLParser must be configured.");
+        } else {
```

Comment:
This shouldn't be in an `else` block. If the condition above proved true, you'd have excepted out of this method.